### DISCUSSION TITLE
The mysterious animated loader bug

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14482] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Loaders cannot properly trigger "OnUse" statuseffects when turrets are shooting. 
Animated ammo belts on loaders rely on a very simple "OnUse" StatusEffect that does not trigger properly on turrets.

- Railgun loader with regular shells properly triggers "OnUse", but Canister shells do not
- Coilgun loader triggers "OnUse" once, then bricks until another box is inserted. 

_My theory:
if projectile inside container inside loader, it activates projectile inside container and not loader 
if projectile inside loader directly it activates loader_

Here is a video, the spark particles are for visual feedback.

https://github.com/user-attachments/assets/d3cc24ab-50a5-4fa1-8080-59303675ffaa

The animated coilgun loader code:

```xml
    <DecorativeSprite name="Ammobelt" texture="Content/Items/Containers/containers.png" depth="0.79" sourcerect="499,663,52,142" origin="0.5,0.5" offset="0,12" offsetanim="Noise" offsetanimspeed="3">
      <IsActiveConditional targetcontaineditem="true" condition="gt 0.0"/>
      <AnimationConditional HasStatusTag="firing" />
    </DecorativeSprite>
    <ItemContainer hideitems="false" drawinventory="true" capacity="1" maxstacksize="1" slotsperrow="6" itempos="82,-270" iteminterval="0,0" itemrotation="0" canbeselected="true" msg="ItemMsgInteractSelect" containedspritedepth="0.79">
      <GuiFrame relativesize="0.18,0.23" anchor="Center" minsize="140,170" maxsize="280,280" style="ItemUI" />
      <StatusEffect type="OnUse" target="Contained" >
        <RequiredItem items="coilgunammo" type="Contained" />
        <Use />
      </StatusEffect>
      <!-- applies "firing" tag that is used by decorativesprite to animate the sprite -->
      <StatusEffect type="OnUse" target="This" tags="firing" duration="0.2" />
      <Containable items="coilgunammo" />
    </ItemContainer>
````

Empty ammo box that works:
The box stops working as soon as you enable spawning coilgunbolts
 
```xml
  <!-- ammo box that doesnt spawn anything and somehow works with animated coilgun loader -->
  <Item name="emptyammobox" identifier="emptycoilgunammobox" scale="0.5" tags="coilgunequipment,coilgunammo,ammobox" category="Weapon" linkable="true" impactsoundtag="impact_metal_heavy">
    <Sprite texture="Content/Items/Containers/containers.png" depth="0.54" sourcerect="800,788,95,78" origin="0.5,0.5" />
    <Body width="90" height="72" density="30" />
    <Holdable canbecombined="true" removeoncombined="false" slots="RightHand,LeftHand" holdpos="0,-30" handle1="0,-30" aimable="false" msg="ItemMsgPickUpSelect">
      <StatValue stattype="MovementSpeed" value="-0.2" />
    </Holdable>
    <ItemContainer hideitems="true" capacity="2" drawinventory="false" canbeselected="false" canbecombined="true" ShowConditionInContainedStateIndicator="true" removecontaineditemsondeconstruct="true" containedstateindicatorstyle="bullet">
      <StatusEffect type="OnUse" target="This" disabledeltatime="true">
        <!-- <SpawnItem identifiers="coilgunbolt" spawnposition="ThisInventory" /> -->
      </StatusEffect>
      <StatusEffect type="OnUse" target="This" condition="-0.5" disabledeltatime="true">
        <!-- <RequiredItem items="coilgunbolt" type="Contained" /> -->
      </StatusEffect>
      <Containable items="coilgunbolt" />
    </ItemContainer>
  </Item>
```

### Reproduction steps

replace coilgun loader code with above

spawn empty box

shoot coilgun with normal box

shoot coilgun with empty box

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Ek-Videogames** [comment]
Yeah saw this when testing some other stuff, I thought I was going crazy that the vanilla coilgun loaders weren't animating...

**Comment by ubertpendragon** [comment]
added a video example
